docs/src/main/asciidoc/dynamodb-guide.adoc

```diff
@@ -0,0 +1,751 @@
+
+|quarkus.dynamodb.credentials.default-provider.reuse-last-provider-enabled
+|Boolean
+|`true`
+|Controls whether the provider should reuse the last successful credentials provider in the chain. Reusing the last successful credentials provider will typically return credentials faster than searching through the chain.
+|===
+
+=== STATIC credentials provider
+If `quarkus.dynamodb.credentials.type=STATIC` is specified, the provider uses the access key & secret access key specified with:
+
+|===
+|Configuration key|Java type|Default Value|Description
+
+|quarkus.dynamodb.credentials.static-provider.access-key-id
+|String
+|
+|AWS access key ID.
+
+|quarkus.dynamodb.credentials.static-provider.secret-access-key
+|String
+|
+|AWS secret access key.
+|===
+
+=== SYSTEM_PROPERTY credentials provider
+If `quarkus.dynamodb.credentials.type=SYSTEM_PROPERTY` is specified, it loads credentials from the `aws.accessKeyId`, `aws.secretAccessKey` and `aws.sessionToken` system properties.
+
+=== ENV_VARIABLE credentials provider
+If `quarkus.dynamodb.credentials.type=ENV_VARIABLE` is specified, it loads credentials from the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN` environment variables.
+
+=== PROFILE credentials provider
+If `quarkus.dynamodb.credentials.type=PROFILE` is specified, credentials are based on AWS configuration profiles.
+This loads credentials from a http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html[profile file], allowing you to share multiple sets of AWS security credentials between different tools like the AWS SDK for Java and the AWS CLI.
+
+It can be configured with:
+|===
+|Configuration key|Java type|Default Value|Description
+
+|quarkus.dynamodb.credentials.profile-provider.profile-name
+|String
+|`default`
+|Define the name of the profile that should be used by this credentials provider. If not specified, the value in `AWS_PROFILE` environment variable or `aws.profile` system property is used.
+|===
+
+=== CONTAINER credentials provider
+If `quarkus.dynamodb.credentials.type=CONTAINER` is specified, it loads credentials from a local metadata service.
+Containers currently supported by AWS SDK:
+- Amazon Elastic Container Service (ECS)
+- AWS Greengrass
+
+
+=== INSTANCE_PROFILE credentials provider
+If `quarkus.dynamodb.credentials.type=INSTANCE_PROFILE` is specified, it loads credentials from the Amazon EC2 Instance Metadata Service.
+
+=== PROCESS credentials provider
+
+If `quarkus.dynamodb.credentials.type=PROCESS` used, credentials to be loaded from an external process. This is used to support the credential_process setting in the profile credentials file.
+See https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#sourcing-credentials-from-external-processes[Sourcing Credentials From External Processes] for more information.
+
+It can be configured with:
```

Comment:
I haven't read everything here. We are working on auto-generating this part from the Javadoc. I don't know how it will fly with this extension as the config is a bit complicated. We'll see.

Reply:
That's good news. So maybe i'll move some of descriptions to the config beans (as here are a little bit more explanations) and remove it from guide. what do you think ?

Reply:
Let's keep it that way for now, we are still making progress on this and I think we will need to test and adjust for this particular doc.

Reply:
ok